### DISCUSSION TITLE
Get full evaluated type annotation

### DISCUSSION THREAD

**Discussion by Dreamsorcerer** [sqlalchemy/sqlalchemy#10204] [Usage Questions] [answered]
Is there any way to get the fully evaluated type from a column (e.g `int | None`)?

Or will I need to do something like:
```
t = c.type.python_type
if c.nullable:
    t = t | None
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

A column does not have a generic argument at runtime, so what you did seems one way.
If you want one from a field of an orm model, you can use the class annotations using the python typing function to obtain them https://docs.python.org/3/library/typing.html?highlight=typing#typing.get_type_hints and then https://docs.python.org/3/library/typing.html?highlight=typing#typing.get_args to get the types from a mapped annotation.

What's the use case?
